test/extended/deployments/util.go

```diff
@@ -424,7 +427,11 @@ func waitForDeployerToComplete(oc *exutil.CLI, name string, timeout time.Duratio
 	}
 	output, err := oc.Run("logs").Args("-f", "pods/"+podName).Output()
 	if err != nil {
-		return "", err
+		// the deployer pod is immediately deleted, so this command can sometimes fail
```

Comment:
deployer pod lifetime between Running and terminal state is artificially prolonged so that should never be the case

Reply:
I don't see that in the code - where?

Reply:
those were the sleep lines you have removed/moved

Reply:
the previous command in this test waits for a deployment to complete.  The sleep lines in the deployment just prolong the deployment.  They don't keep the pod around longer after the deployment is completed.  Once the deployment completes, this call races against the deployment controller to delete the deployer pod, and we only win ~4/5 times.